agentos/agent_output.py

```diff
@@ -25,6 +25,30 @@
 
 RunStats = namedtuple("RunStats", _RUN_STATS_MEMBERS)
 
+SPEC_ATTRS = []
```

Comment:
I know you mentioned this in your comment, but is there any reason these couldn't be `@classmethod`s?

Reply:
Yep, I was able to move these functions into the classes.